app/controllers/api/v1/threat_levels_controller.rb

```diff
@@ -17,44 +17,83 @@ def set_limit
         @limit = @limit > 0 ? @limit : 5
       end
 
-      def threat_levels_data
-        {threat_levels: threat_levels_for_profiles}
-      end
-
       def threat_levels_for_profiles
-        Profile.all.collect { |p| threat_levels_for_profile(p) }
+        @assessments_data = {}
+        @threat_assessments = []
+        threat_assessment_ids = ThreatAssessment.find_by_sql(sql).collect {|ta| ta.id}
+        if threat_assessment_ids.present?
+          results_data = build_results_data(threat_assessment_ids)
+          fill_threat_assessments_data(results_data)
+        end
+        @assessments_data.values
       end
 
-      def threat_levels_for_profile(profile)
-        {
-          profile_id: profile.id,
-          threat_assessments: threat_assessments_for_profile(profile),
-          threat_level_assessment_runs: ThreatLevel::AssessmentRunsDataBuilderService.get(profile)
-        }
-      end
+      def fill_threat_assessments_data(results_data)
+        @threat_assessments.uniq!
```

Comment:
Kind of awkward. How about `@threat_assessments << ta unless @threat_assessments.include? ta` further on down (in `build_results_data:55`?